tests/self/basic.bt

```diff
@@ -0,0 +1,117 @@
+test:increment_decrement_variable {
+  $x = 10;
+  $result1 = $x++;
+  $result2 = ++$x;
+  $result3 = $x--;
+  $result4 = --$x;
+  if ($result1 != 10 || $result2 != 12 || $result3 != 12 || $result4 != 10) {
+    return 1;
+  }
+}
+
+test:variable_strings_are_memset {
+  $x = "xxxxx";
+  $x = "a";
+  @vs[$x] = 1;
+  $y = "yyyyy";
+  $y = "a";
+  @vs[$y] = 1;
+  if (len(@vs) != 1) {
+    return 1;
+  }
+  clear(@vs);
+}
+
+test:map_strings_are_memset {
+  @x = "xxxxx";
+  @x = "a";
+  @ms[@x] = 1;
+  @y = "yyyyy";
+  @y = "a";
+  @ms[@y] = 1;
+  if (len(@ms) != 1) {
+    return 1;
+  }
+  clear(@ms);
+  clear(@x);
+  clear(@y);
+}
+
+test:print_large_int {
+  $a = 10223372036854775807;
+  if ($a != 10223372036854775807) {
+    return 1;
+  }
+}
+
+test:block_expression {
+  let $a = { let $b = 1; $b };
+  if ($a != 1) {
+    return 1;
+  }
+}
+
+test:print_block_expression_with_print {
+  $result = { $x = 1; $x > 1 };
+  if ($result != false) {
+    return 1;
+  }
+}
+
+test:ppid {
+  if (ppid(curtask) <= 0) {
+    return 1;
+  }
+}
+
+test:strerror {
+  if (strerror(22) != "Invalid argument") {
+    return 1;
+  }
+}
+
+test:comm_from_pid {
+  if (comm(pid) != "bpftrace") {
+    return 1;
+  }
+}
+
+test:clear_map {
+  @ = 1;
+  @a[1] = 1;
+  clear(@);
+  clear(@a);
+}
+
+test:delete_map {
```

Comment:
These are partially testing delete - we should use it in tandem with `has_key` or a map lookup